### ISSUE TITLE
Remove or rename invalid CSS properties caused by using mixins

### ISSUE THREAD

**Issue by jroebu14** [styled-components/stylelint-processor-styled-components#271]
Is it possible to prevent outputting invalid CSS properties that would fail a CSS syntax validation plugin such as [csstree/stylelint-validator](https://github.com/csstree/stylelint-validator).

An example of this would be when using a mixin:

```js
const example = css`
${({ height }) => height ? `height: ${height};` : ''}
`;
```

which the processor outputs

```js
const example = css`
${({ height }) => height ? `height: ${height};` : ''}
width: 100%;
`;
```

```js
```

**Comment by leoyli** [comment]
Yes, I have also received this `selector-type-no-unknown` error...

**Comment by chinesedfan** [comment]
> My suggestion would be to wrap -styled-mixin0: dummyValue; in a CSS comment or prefix it with one more dash to turn it into a custom property e.g. --styled-mixin0: dummyValue;

@jroebu14 Thanks for your suggestions. Sorry we didn't find a perfect solution. Comments may also introduce comment-related false-positive errors. Adding one more dash can't solve the problem from the root, either. And it can also break something like @leoyli's workaround.

Currently, @leoyli's workaround is one of the best ideas .

**Comment by leoyli** [comment]
@chinesedfan, can be considered to add my workaround into the recommended setup? I mean people using it expect it can also take care about all valid use case.

**Comment by jroebu14** [comment]
@leoyli @chinesedfan thanks for responding to my issue. Leoyli's is a good solution but unfortunately it won't work in our specific case where we're also using the [csstree/stylelint-validator](https://github.com/csstree/stylelint-validator) plugin. This plugin has it's own ignore option which doesn't support regular expressions. I have [suggested this functionality in csstree/stylelint-validator](https://github.com/csstree/stylelint-validator/issues/19).

For now [my workaround looks like this](https://github.com/bbc/psammead/blob/latest/lib/stylelint/processorRemoveMixins.js). As you can see it's not ideal but it works for us in the short term.

**Comment by PierreGUI** [comment]
Hi, I'm running into the same issue with `stylelint-order`.  
Related issue: https://github.com/hudochenkov/stylelint-order/issues/63

#### Suggestion:  
What about adding a configuration to give a custom name for mixins. Default would still be `-styled-mixin${id}`, but in some cases we could make it whatever we want (such as `--styled-mixin` or `zz-styled-mixin` in my case to solve sorting ;) )

Thanks
